### PR TITLE
Storage

### PR DESCRIPTION
**Pull request by gasangw**
- Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)
- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -36,14 +42,16 @@ def create_person
...
       student_permission = true if provided_permission == 'Y'
       student_permission = false if provided_permission == 'N'
-      @people.push(Student.new(age, name, parent_permission: student_permission))
+      @people.push(Student.new(age, student_permission, name))
+
     when 2
       print 'Specialization: '
       specialization = gets.chomp.to_s
-      @people.push(Teacher.new(age, name, specialization))
+      @people.push(Teacher.new(specialization, age, name))
```
Comment (GzimAsani) [inline app.rb:54]: - [x] When I register one teacher after I leave the app the teacher's name is not present  on the JSON file. The other objects like specialization and age are but name not. Kindly check into that

Comment (GzimAsani) [reply app.rb:54]: - [x] People's ID's are not showing here, kindly update it 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello **TEAM**           :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your tests, you did most of the work +1: 
 
 ## Required Changes 

- [ ]  On your readme file, please include a section explaining how to clone the project
- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by petushka1** [comment]
## Fixes
- [x] fix teacher class to save 'name' properly
- [x] extend README with detailed instructions

**Review by smunozmo — APPROVED** [review]
Hi @gasangw, 👋 

## Highlights 👏 

- Excellent job making all the suggested changes! 🎉 
- Data is preserved properly, perfect! 😎 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

# APPROVED! 💯  💯  💯 




Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**